### PR TITLE
[WIP] Apply Doctrine CS

### PR DESCRIPTION
**Pull request by Majkl578**
Initial import of Doctrine CS.

@goetas Excluded some rules you mentioned before that you don't like.

Also excluded BC-breaking rules since the code base is currently a bit messy, a lot of inconsistent/missing parameter/return types. :(

Only applied to src/ to give you some look at the result, and to have something to discuss/improve later on.

### CODE REVIEW

**Comment by goetas** [comment]
Is there a way to disable the `=` alignment?

File: src/EventDispatcher/EventDispatcherInterface.php [hunk 1]
```diff
@@ -23,31 +19,18 @@ public function hasListeners(string $eventName, string $class, string $format):
...
      * were added to the dispatcher.
      *
-     * @param string $eventName
-     * @param string $class
-     * @param string $format
-     * @param Event $event
-     * @return void
      */
-    public function dispatch($eventName, string $class, string $format, Event $event): void;
+    public function dispatch(string $eventName, string $class, string $format, Event $event): void;
```
Comment (goetas) [inline src/EventDispatcher/EventDispatcherInterface.php]: good catch, this is a bug! :)

**Comment by Majkl578** [comment]
> Is there a way to disable the `=` alignment?

yes

> Why remove the `@author` annotation?

It's totally useless and very often outdated, Git already provides more accurate information.
See https://github.com/doctrine/coding-standard/pull/9#discussion_r137123497 + https://github.com/doctrine/coding-standard/pull/9#issuecomment-353438394

File: src/EventDispatcher/EventDispatcherInterface.php (hunk 1 above)
Comment (Majkl578) [reply src/EventDispatcher/EventDispatcherInterface.php]: There is more of missing types, some break more things so I would probably not touch types in first iteration and do them separately, then add proper docblocks for properties etc. (this is especially in extremely bad shape in Serializer).
Then we can also consider PHPStan.

**Comment by Majkl578** [comment]
Also please decide the following:
* yoda conditions: enforce or forbid
* assignments in conditions: allow or not

**Comment by goetas** [comment]
How to run the comment to apply the rules? is it documented somewhere?

**Comment by Majkl578** [comment]
https://www.doctrine-project.org/projects/doctrine-coding-standard/en/latest/reference/index.html#composer-project-dependency

Since the project has `phpcs.xml.dist`, it's enough to simply run:
* `vendor/bin/phpcs` to validate CS
* `vendor/bin/phpcbf` to fix fixable errors according to CS

**Comment by goetas** [comment]
will try, tnx!

**Comment by goetas** [comment]
implemented in https://github.com/schmittjoh/serializer/pull/971

**Pull request closed without merging**